src/DoctrineAuditBundle/Manager/AuditManager.php

```diff
@@ -358,4 +362,41 @@ private function selectStorageSpace(EntityManagerInterface $em): EntityManagerIn
     {
         return $this->configuration->getEntityManager() ?? $em;
     }
+
+    /**
+     * @param AuditReader            $reader
+     * @param EntityManagerInterface $em
+     * @param string                 $hash
+     * @param string                 $field
+     *
+     * @throws AccessDeniedException
+     * @throws InvalidArgumentException
+     *
+     * @return null|object
+     */
+    public function revert(AuditReader $reader, EntityManagerInterface $em, string $hash, string $field)
+    {
+        $current_audit = $reader->getAuditsByTransactionHash($hash);
+        // Get the fully qualified class name
+        $entity_fqcn = array_keys($current_audit);
+        $entity_fqcn = reset($entity_fqcn);
+        /** @var AuditEntry $audited_entry */
+        $audited_entry = $current_audit[$entity_fqcn];
+        $audited_entry = reset($audited_entry);
+
+        // get real entity
+        $original_entity = $em->getRepository($entity_fqcn)->find($audited_entry->getObjectId());
+
+        // Get all differences
+        $diffs = $audited_entry->getDiffs();
+        // get field value to revert
+        $field_value = $diffs[$field]['old'];
+
+        $field = ucfirst(strtolower($field));
+        $setMethod = "set{$field}";
```

Comment:
This won't work depending on how you entity properties are named. For example, if a property is using snake case (eg. `created_at`) but the relevant accessors are using camel case (eg. `setCreatedAt` and `getCreatedAt`)

Reply:
This can be change with `str_replace('_', '', ucwords(strtolower($field), '_'))`